osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -345,9 +320,10 @@ private void processBeatmapChanges()
         {
             try
             {
-                foreach (var set in setsRequiringRemoval) removeBeatmapSet(set);
+                // TODO: chekc whether we still need beatmap sets by ID
```

Comment:
Should this todo be here?

Reply:
It shouldn't.

I checked on this. We kinda need this stuff locally to ensure we're handling the batch application correctly (see the schedule related comment above). If it wasn't for this we can simplify this class a lot more.

Probably a direction for the future.